usecases/encryption-at-rest.markdown

```diff
@@ -0,0 +1,53 @@
+with file system access, is able to read the contents of the records.
+
+This present a challenge for an Enterprise using Kafka to distribute confidential data or PII (Personally Identifiable
+Information) between their applications.  The problem is made more complex if the organization has opted to utilise a
+Cloud Kafka Service as the worry now extends to the Kafka Adminsitrators in the service providers organization.
+
+### Isn't TLS sufficient?
+
+TLS encrypts the content as it traverses the wire.  It means that someone using a network sniffer cannot tap what is being
+sent over the wire between the application and the Kafka Broker.  However, once the network packets arrive at the broker,
+the packets are decrypted and exist in the clear once again.  This means the records are in the clear in the memory  of the 
+broker and in the clear when the data is written to the disk.
+
+TLS does not change the problem.
+
+### Isn't storage volume encryption an answer?
+
+With storage volume encryption, the contents of the device are encrypted with a key.  This approach provides some mitigations.
+If the storage device is stolen or the storage device attached to an attacker's computer, the attacker won't have the encryption
+key so won't be able to read the data, including the Kafka records.
+
+However, there are shortcomings.  When an encrypted storage volume is mounted to a computer the encryption keys are applied at
+operating system level meaning the volume's contents appear uncrypted.  Anybody with shell level access is likely to be able
+to read the data, including the Kafka records.  Storage volume encryption doesn't really solve the problem.
+
+### Can't the applications encrypt/decrypt the data?
+
+It is possible for producing applications to encrypt data before sending it to Kafka, and for consuming applications to decrypt it
+again.  Indeed there are some libraries out in the Kafka ecosystem that help an application do just that.   With this approach
+the Kafka Broker never possess the record in the clear, nor does it ever possess the encryption keys need to decrypt the data.
+This approach does offer a solution to the problem.
+
+The disadvantages to this approach are two-fold.
+
+* Firstly, there is a key distribution problem - something needs to get the encryption keys to the applications and manage events
+such as key rotation.  This is a burden on the application teams.
+* Secondly, the Kafka client ecosystem is a polyglot space : Java, Go, Python, Javascript, Rust.  It is common to see an organization
+with application written in many different languages.  This approach means the encryption/decryption imeplementations needs to exist
+in those same langauges and the implementations need to interoperate.  This is too is a burden on the application development teams.
+What's more - writing encryption/decryption code is a skilled job.
+
+
+## Kroxylicious Topic Encryption
```

Comment:
I think we want to contrast APP level encryption with Proxy level either directly in above bullet or before we get to the Kroxylicious solution.